networking/change-block-size.md

```diff
@@ -0,0 +1,303 @@
+
+We add a new IPPool with the CIDR range, 10.0.0.0/16.
+
+Create a new-pool.yaml.
+
+<pre>
+apiVersion: projectcalico.org/v3
+kind: IPPool
+metadata:
+  name: new-pool
+spec:
+  cidr: 10.0.0.0/16
+  ipipMode: Always
+  natOutgoing: true
+</pre>
+
+Apply the changes.
+
+```
+calicoctl apply -f new-pool.yaml
+```
+
+Let’s verify the new IP pool.
+
+```
+calicoctl get ippool -o wide
+```
+
+<pre>
+NAME                  CIDR             NAT    IPIPMODE   DISABLED
+default-ipv4-ippool   192.168.0.0/16   true   Always     false
+new-pool              10.0.0.0/16      true   Always     false
+</pre>
+
+#### Step 2: Disable the old IP pool
+
+List the existing IP pool definition.
+
+```
+calicoctl get ippool -o yaml --export > pool.yaml
+```
+
+<pre>
+apiVersion: projectcalico.org/v3
+items:
+- apiVersion: projectcalico.org/v3
+  kind: IPPool
+  metadata:
+    name: default-ipv4-ippool
+  spec:
+    cidr: 192.0.0.0/16
+    ipipMode: Always
+    natOutgoing: true
+- apiVersion: projectcalico.org/v3
+  kind: IPPool
+  metadata:
+    name: new-pool
+  spec:
+    cidr: 10.0.0.0/16
+    ipipMode: Always
+    natOutgoing: true
+</pre>
+
+Edit pool.yaml and disable this IP pool by setting: `disabled: true`
+
+<pre>
+apiVersion: projectcalico.org/v3
+kind: IPPool
+metadata:
+  name: default-ipv4-ippool
+spec:
+  cidr: 192.0.0.0/16
+  ipipMode: Always
+  natOutgoing: true
+  disabled: true
+</pre>
+
+Remember, disabling a pool affects only new IP allocations; networking for existing pods is not affected.
+
+Apply the changes.
+
+```
+calicoctl apply -f pool.yaml
+```
+Verify the changes.
+
+```
+calicoctl get ippool -o wide
+```
+
+<pre>
+NAME                  CIDR             NAT    IPIPMODE   DISABLED
+default-ipv4-ippool   192.168.0.0/16   true   Always     true
+new-pool              10.0.0.0/16      true   Always     false
+</pre>
+
+#### Step 3: Delete pods from the old IP pool
+
+Next, delete all of the existing pods from the old IP pool. (In our example, **coredns** is our only pod; for multiple pods you would trigger a deletion for all pods in the cluster.)
+
+```
+kubectl delete pod -n kube-system coredns-6f4fd4bdf-8q7zp
+```
+Restart all pods with just one command. 
+
+> **WARNING!** This is disruptive and may take several minutes depending on the number of pods deployed.
+{: .alert .alert-danger}
+
+```
+kubectl delete pod -A --all
+```
+
+#### Step 4: Delete the old IP pool
+
+Now that you’ve verified that pods are getting IPs from the new range, you can safely delete the old pool.
+
+```
+calicoctl delete pool default-ipv4-ippool
+```
+
+#### Step 5: Create a new pool with custom CIDR block
+
+<pre>
+apiVersion: projectcalico.org/v3
+kind: IPPool
+metadata:
+  name: default-ipv4-ippool
+spec:
+  blockSize: 28
+  cidr: 192.0.0.0/16
+  ipipMode: Always
+  natOutgoing: true
+</pre>
+
+Apply the changes.
+
+```
+calicoctl apply -f pool.yaml
+```
+
+#### Step 6: Disable the temporary IP pool
+
+```
+calicoctl get ippool -o yaml --export > pool.yaml
```

Comment:
Can use the same patch from above to disable here as well, but swapping out the pool name in the command.